hack/tools/kapinger/Dockerfile.windows

```diff
@@ -0,0 +1,16 @@
+FROM --platform=linux/amd64 mcr.microsoft.com/oss/go/microsoft/golang:1.22 AS builder
+
+WORKDIR /build
+COPY . .
+ENV GOOS=windows
+ENV GOARCH=amd64 
+
+RUN echo "building kapinger for OS: $GOOS, ARCH: $GOARCH"
+RUN go mod download
+RUN ls
+RUN CGO_ENABLED=0 go build -o kapinger.exe .
+
+FROM --platform=windows/amd64 mcr.microsoft.com/windows/nanoserver:ltsc2022
```

Comment:
everything except the final stage is identical, use targets and fold in to single dockerfile?

Reply:
yeah tbh I was pretty focused on how to get a target to enumerate 1:1 with `linux/amd64,windows/amd64` in the platform list, then specify final nanoserver/scratch image as targets, but it didn't seem possible so avoided that

but then found `provenance=false` doesn't create a manifest, so we can use imagetools and stitch together later after building in separate jobs, which yeah now we can add target back to [the action](https://github.com/docker/build-push-action#inputs)